### DISCUSSION TITLE
火狐显示异常

### DISCUSSION THREAD

**Discussion by 1332977811** [Tencent/libpag#1663] [General] [answered]
采用官方兼容解码示例代码测试，在火狐浏览器上依然无法显示动态效果，并报错
参考代码地址：https://github.com/libpag/pag-web/blob/main/pages/software-decoder.html
![image](https://github.com/Tencent/libpag/assets/57947126/a9902c90-a887-440f-9d82-641d4f389007)

![image](https://github.com/Tencent/libpag/assets/57947126/25b2e63d-74fe-453e-8404-b496cb168488)

**Comment by zhalice2011** [comment, marked as answer]
你好, 可以看看这个兼容性文档: https://github.com/Tencent/libpag/blob/main/web/doc/compatibility.md

Firefox 只能播放全矢量的PAG, 比如PAG官网首页的这个https://pag.art/

<img width="1038" alt="image" src="https://github.com/Tencent/libpag/assets/24289349/884445a0-7479-4ccc-96dd-257048d58b8e">

**Comment by 1332977811** [comment]
可做参考
https://github.com/libpag/ffavc/blob/main/web/demo/index.ts
`
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta
      http-equiv="X-UA-Compatible"
      content="IE=edge"
    />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1.0"
    />
    <title>PAG</title>
    <link
      rel="icon"
      href="https://pag.io/img/favicon.png"
    />
    <style>
      body {
        padding: 12px;
      }
      .header {
        height: 64px;
        border-bottom: 1px solid rgb(193, 193, 193);
        margin-bottom: 24px;
      }
    </style>
  </head>
  <body>
    <div>
      <canvas id="pag"></canvas>
    </div>
    <script src="https://unpkg.com/libpag@0.1.7-alpha/lib/libpag.min.js"></script>
    <script src="https://unpkg.com/ffavc@latest/lib/ffavc.min.js"></script>
    <script>
      window.onload = async () => {
        const FFAVC = await window.ffavc.FFAVCInit({})
        console.log('FFAVC', FFAVC)

        const ffavcDecoderFactory = new FFAVC.FFAVCDecoderFactory()

        const PAG = await window.libpag.PAGInit()
        PAG.registerSoftwareDecoderFactory(ffavcDecoderFactory)

        const buffer = await fetch('本地文件地址').then(
          (response) => response.arrayBuffer()
        )
        const pagFile = await PAG.PAGFile.load(buffer)

        const canvas = document.getElementById('pag')
        canvas.width = pagFile.width()
        canvas.height = pagFile.height()

        let pagView = await PAG.PAGView.init(pagFile, canvas)
        pagView.setRepeatCount(0)
        await pagView?.play()
      }
    </script>
  </body>
</html>

`
